### PR TITLE
CollectTest v2 🎉

### PR DESCRIPTION
**Pull request by dorschw**
- replacing v1 with v2
- `relative_to_packs` is no longer static, allowing for custom content paths
- do install pack when test is invalid 
- remove `run_nightly` argument (as it does not have any effect)
- add content path to error messages, rather than just the id
- improve git diff parsing:
  - allow 3-part diff lines
  - allow non-A/M/D status
- get branch name from git util 
- improved [file-ignoring](https://github.com/demisto/content/pull/20731/files#diff-c697d5816da92fa8279640fcd77747be08c06141e35fca6fcaf70a6367298489R47-R54) mechanism 
- add v6.9

### CODE REVIEW

File: Tests/scripts/collect_tests/path_manager.py [hunk 1]
```diff
@@ -26,7 +28,12 @@ class PathManager:
 
     def __init__(self, content_path: Path):
         self.content_path = content_path
-        self.content_repo = Repo(content_path)
+        try:
+            self.content_repo = Repo(content_path)
+        except git.exc.InvalidGitRepositoryError:
+            if not os.getenv('UNIT_TESTING'):
+                raise
+            self.content_repo = None
```
Comment (dorschw) [inline Tests/scripts/collect_tests/path_manager.py:36]: @DeanArbel real change

File: Tests/scripts/collect_tests/path_manager.py [hunk 2]
```diff
@@ -37,9 +44,9 @@ def __init__(self, content_path: Path):
 
         self.id_set_path = PathManager.ARTIFACTS_PATH / 'id_set.json'
         self.conf_path = PathManager.ARTIFACTS_PATH / 'conf.json'
-        self.output_tests_file = PathManager.ARTIFACTS_PATH / 'filter_file_v2.txt'
-        self.output_packs_file = PathManager.ARTIFACTS_PATH / 'content_packs_to_install_v2.txt'
-        self.output_machines_file = PathManager.ARTIFACTS_PATH / 'filter_envs_v2.json'
+        self.output_tests_file = PathManager.ARTIFACTS_PATH / 'filter_file.txt'
+        self.output_packs_file = PathManager.ARTIFACTS_PATH / 'content_packs_to_install.txt'
+        self.output_machines_file = PathManager.ARTIFACTS_PATH / 'filter_envs.json'
```
Comment (dorschw) [inline Tests/scripts/collect_tests/path_manager.py:49]: .

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 3]
```diff
@@ -1531,12 +1531,12 @@ def create_test_file(is_nightly, skip_save=False, path_to_pack='', marketplace_v
...
-        with open(os.path.join(ARTIFACTS_FOLDER, 'filter_file.txt'), 'w') as filter_file:
+        logging.info("Creating filter_file_old.txt")
+        with open(os.path.join(ARTIFACTS_FOLDER, 'filter_file_old.txt'), 'w') as filter_file:
             filter_file.write(tests_string)
-        # content_packs_to_install.txt is not used in nightly build
-        logging.info("Creating content_packs_to_install.txt")
-        with open(os.path.join(ARTIFACTS_FOLDER, 'content_packs_to_install.txt'), 'w') as content_packs_to_install:
+        # content_packs_to_install_old.txt is not used in nightly build
+        logging.info("Creating content_packs_to_install_old.txt")
+        with open(os.path.join(ARTIFACTS_FOLDER, 'content_packs_to_install_old.txt'), 'w') as content_packs_to_install:
```
Comment (dorschw) [inline Tests/scripts/collect_tests_and_content_packs.py:1539]: .

File: Tests/scripts/infrastructure_tests/test_collect_tests.py [hunk 4]
```diff
@@ -1,18 +1,21 @@
...
+from Tests.scripts.collect_tests import collect_tests
+# importing Machine,FileType from collect_tests (rather than utils) to compare class member values
+from Tests.scripts.collect_tests.collect_tests import (
+    BranchTestCollector, FileType, Machine, XSIAMNightlyTestCollector,
+    XSOARNightlyTestCollector)
 from Tests.scripts.collect_tests.constants import XSOAR_SANITY_TEST_NAMES
 from Tests.scripts.collect_tests.path_manager import PathManager
 from Tests.scripts.collect_tests.utils import PackManager
 
+os.environ['UNIT_TESTING'] = 'True'
```
Comment (dorschw) [inline Tests/scripts/infrastructure_tests/test_collect_tests.py:17]: .

File: Tests/scripts/collect_tests/utils.py [hunk 5]
```diff
@@ -76,6 +77,7 @@ class Machine(Enum):
     V6_5 = Version('6.5')
     V6_6 = Version('6.6')
     V6_8 = Version('6.8')
+    V6_9 = Version('6.9')
```
Comment (dorschw) [inline Tests/scripts/collect_tests/utils.py:80]: .

File: Tests/scripts/collect_tests/path_manager.py (hunk 1 above)
Comment (dorschw) [inline Tests/scripts/collect_tests/path_manager.py:36]: .

**Review by dorschw — COMMENTED** [review]
@DeanArbel see the commented parts - the rest is just noise from files being moved under `infrastucture_tests`

**Review by DeanArbel — APPROVED** [review]

**Pull request merged**